agent/metrics/instancehealth/instancehealth.go

```diff
@@ -0,0 +1,56 @@
+package instancehealth
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/aws/amazon-ecs-agent/agent/api/errors"
+)
+
+// genericCounter records call and error counts of API calls
+type genericMetric struct {
+	calls        int64
+	errors       int64
+	errorMessage atomic.Value
+}
+
+var DockerMetric *genericMetric
+
+func init() {
+	DockerMetric = &genericMetric{}
+}
+
+func (gc *genericMetric) IncrementCallCount() {
+	atomic.AddInt64(&gc.calls, 1)
+}
+
+func (gc *genericMetric) GetCallCount() int64 {
+	return atomic.LoadInt64(&gc.calls)
+}
+
+func (gc *genericMetric) IncrementErrorCount() {
+	atomic.AddInt64(&gc.errors, 1)
+}
+
+func (gc *genericMetric) GetErrorCount() int64 {
+	return atomic.LoadInt64(&gc.errors)
+}
+
+func (gc *genericMetric) RecordErrorMessage(errorMessage string) {
```

Comment:
I would suggest making this an internal function if it doesnt need to be part of the public interface at the moment. Maybe it doesnt even need to be a function separate from RecordError?

Reply:
Agreed and updated.